subprojects/logging/src/main/java/org/gradle/internal/featurelifecycle/LoggingDeprecatedFeatureHandler.java

```diff
@@ -115,14 +115,23 @@ private static void appendLogTraceIfNecessary(List<StackTraceElement> stack, Str
             return;
         }
 
+        boolean nonSystemOrScriptElementFound = false;
         for (StackTraceElement element : stack) {
             if (isGradleScriptElement(element)) {
                 // only print first Gradle script stack trace element
                 appendStackTraceElement(element, message, lineSeparator);
-                appendRunWithStacktraceInfo(message, lineSeparator);
-                return;
+                break;
+            }
+
+            if (!nonSystemOrScriptElementFound && !isSystemElement(element)) {
+                // only print first non Gradle script and non-system stack trace element
+                appendStackTraceElement(element, message, lineSeparator);
+                nonSystemOrScriptElementFound = true;
             }
         }
+        if (!stack.isEmpty()) {
```

Comment:
I would rather do this: 

```
        boolean nonSystemOrScriptElementFound = false;
        boolean scriptElementFound = false;
        for (StackTraceElement element : stack) {
            if (isGradleScriptElement(element)) {
                // only print first Gradle script stack trace element
                appendStackTraceElement(element, message, lineSeparator);
                scriptElementFound = true;
                break;
            }

            if (!nonSystemOrScriptElementFound && !isSystemElement(element)) {
                // only print first non Gradle script and non-system stack trace element
                appendStackTraceElement(element, message, lineSeparator);
                nonSystemOrScriptElementFound = true;
            }
        }
        if (scriptElementFound || nonSystemOrScriptElementFound) {
            appendRunWithStacktraceInfo(message, lineSeparator);
        }
```

Reply:
Actually, that was intentional. I first had it similar to what you have, just with the one boolean already existing. But I thought if there is any stack trace, it should always show the hint. This way it is up to the user to decide whether any stack frame is helpful or not. Currently there are also helpful stack frames and Gradle showing nothing. :-)

But if you really prefer to not show it, I can change it of course.